stl.go

```diff
@@ -587,23 +623,57 @@ type ttiBlock struct {
 func newTTIBlock(i *Item, idx int) (t *ttiBlock) {
 	// Init
 	t = &ttiBlock{
-		commentFlag:          stlCommentFlagTextContainsSubtitleData,
-		cumulativeStatus:     stlCumulativeStatusSubtitleNotPartOfACumulativeSet,
-		extensionBlockNumber: 255,
-		justificationCode:    stlJustificationCodeLeftJustifiedText,
-		subtitleGroupNumber:  0,
-		subtitleNumber:       idx,
-		timecodeIn:           i.StartAt,
-		timecodeOut:          i.EndAt,
-		verticalPosition:     20,
+		subtitleNumber: idx,
+		timecodeIn:     i.StartAt,
+		timecodeOut:    i.EndAt,
+	}
+
+	if i.Metadata.STLCommentFlag != nil {
+		t.commentFlag = *i.Metadata.STLCommentFlag
+	} else {
+		t.commentFlag = stlCommentFlagTextContainsSubtitleData
+	}
+
+	if i.Metadata.STLCumulativeStatus != nil {
+		t.commentFlag = *i.Metadata.STLCumulativeStatus
+	} else {
+		t.commentFlag = stlCumulativeStatusSubtitleNotPartOfACumulativeSet
+	}
+
+	if i.Metadata.STLExtensionBlockNumber != nil {
+		t.extensionBlockNumber = *i.Metadata.STLExtensionBlockNumber
+	} else {
+		t.extensionBlockNumber = 255
+	}
+
+	if i.InlineStyle.STLJustificationCode != nil {
+		t.justificationCode = byte(*i.InlineStyle.STLJustificationCode)
+	} else {
+		t.justificationCode = stlJustificationCodeLeftJustifiedText
+	}
+
+	if i.Metadata.STLSubtitleGroupNumber != nil {
+		t.subtitleGroupNumber = *i.Metadata.STLSubtitleGroupNumber
+	} else {
+		t.subtitleGroupNumber = 0
+	}
+
+	if i.InlineStyle.STLVerticalPostion != nil && *i.InlineStyle.STLVerticalPostion > 0 {
+		t.verticalPosition = int(*i.InlineStyle.STLVerticalPostion)
+	} else {
+		t.verticalPosition = 20
 	}
 
 	// Add text
-	var lines []string
-	for _, l := range i.Lines {
-		lines = append(lines, l.String())
+	if i.Metadata.STLText != nil && len(*i.Metadata.STLText) > 0 {
+		t.text = *i.Metadata.STLText
+	} else {
+		var lines []string
+		for _, l := range i.Lines {
+			lines = append(lines, l.String())
+		}
+		t.text = []byte(strings.Join(lines, "\n"))
```

Comment:
Replace this line by

```go
t.text = []byte(encodeTextSTL(strings.Join(lines, "\n")))
```

and check tests still pass

Reply:
Thanks. done